docs/source/modules.rst

```diff
@@ -0,0 +1,146 @@
+the same pattern as much as possible.
+
+The API of each module is composed of two parts. The *auth* parameter contains
+the pieces of information that are related to the Sensu Go backend that the
+module is connecting to. All other parameters hold the information related to
+the resource that we are operating on. Typical task would look like this:
+
+.. code-block:: yaml
+
+   - name: Make sure asset is present
+     asset:
+       auth:
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+       name: my-asset-name
+       builds: ...
+
+Almost every module for manipulating resources has its info counterpart that
+can be used to retrieve information about the certain resources. For example,
+to get a list of assets from the ``my-namespace`` namespace, we would write a
+task like this one:
+
+.. code-block:: yaml
+
+   - name: Fetch list of all assets
+     asset:
+       auth:
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+     register: result
+
+We can also retrieve information about a single asset by adding a *name*
+parameter to the previous task:
+
+.. code-block:: yaml
+
+   - name: Fetch a specific asset
+     asset:
+       auth:
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+       name: my-asset-name
+     register: result
+
+Info modules always return a list of objects that Sensu API returned. And if
+we try to fetch a non-existing resource, the result will hold an empty list.
+This makes it easy to write a conditional tasks using next pattern:
+
+.. code-block:: yaml
+
+   - name: Fetch a specific asset
+     asset:
+       auth:
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+       name: my-asset-name
+     register: result
+
+   - name: Do something if asset is there
+     debug:
+       msg: We are doing somethin
+     when: result.objects | length == 1
+
+Of course, you can also loop over the result using a *loop* construct:
+
+.. code-block:: yaml
+
+   - name: Fetch list of all assets
+     asset:
+       auth:
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+     register: result
+
+   - name: Display number of builds in an asset
+     debug:
+       msg: "{{ item.metadata.name }}: {{ item.builds | length }}"
+     loop: result.objects
+
+If we would like to remove a certain resource from the Sensu backend, we can
+write a task and set its *state* parameter to ``absent``:
+
+.. code-block:: yaml
+
+   - name: Make sure asset is absent
+     asset:
+       auth:
+         url: http://sensu.host.com:8080
+         user: my-user
+         password: pass
+       namespace: my-namespace
+       name: my-asset-name
+       state: absent
```

Comment:
Can we move this part right after the example that demonstrates creation of an asset? I think it's easier to follow along if we first show a couple of tasks that manipulate resources using _asset_, and then a couple of tasks just fetching info via the _asset_info_ module. Otherwise one gets confused easily.